### PR TITLE
format code with gofumpt

### PR DESCRIPTION
**Pull request by thaJeztah**
gofumpt (https://github.com/mvdan/gofumpt) provides a supserset of `gofmt` / `go fmt`, and addresses various formatting issues that linters may be checking for.

We can consider enabling the `gofumpt` linter to verify the formatting in CI, although not every developer may have it installed, so for now this runs it once to get formatting in shape.

### CODE REVIEW

**Comment by milosgajdos** [comment]
I'm pretty sure if we bump the linter to `1.50+` (https://github.com/distribution/distribution/blob/7f9f86c4116f83e99f0d461d4f9b3b41569dc7c1/dockerfiles/lint.Dockerfile#L5) it will call us out on this -- I saw that happening to me in other codebases 😅

**Review by milosgajdos — APPROVED** [review]

**Comment by thaJeztah** [comment]
Pushed a couple of manual fixes after I ran gofumpt (had those in another branch I was working on, but this PR was a better fit). https://github.com/distribution/distribution/compare/bcb528766fcbac2f52190e712008ce175b0b4eb1..ce252e8ebf1608aed96bc2c6352da566dccd3b1a

File: registry/storage/driver/s3-aws/s3.go [hunk 1]
```diff
@@ -1166,16 +1165,11 @@ func (d *driver) doWalk(parentCtx context.Context, objectCount *int64, path, pre
...
-// Eg 1 directoryDiff("/path/to/folder", "/path/to/folder/folder/file")
-//   => [ "/path/to/folder/folder" ],
-// Eg 2 directoryDiff("/path/to/folder/folder1", "/path/to/folder/folder2/file")
-//   => [ "/path/to/folder/folder2" ]
-// Eg 3 directoryDiff("/path/to/folder/folder1/file", "/path/to/folder/folder2/file")
-//  => [ "/path/to/folder/folder2" ]
-// Eg 4 directoryDiff("/path/to/folder/folder1/file", "/path/to/folder/folder2/folder1/file")
-//   => [ "/path/to/folder/folder2", "/path/to/folder/folder2/folder1" ]
-// Eg 5 directoryDiff("/", "/path/to/folder/folder/file")
-//   => [ "/path", "/path/to", "/path/to/folder", "/path/to/folder/folder" ],
```
Comment (deleteriousEffect) [inline registry/storage/driver/s3-aws/s3.go:1178]: This formatting got manged a bit here, can we keep the original?

File: registry/storage/paths.go [hunk 2]
```diff
@@ -23,25 +23,33 @@ const (
...
+//	        │               └── <algorithm>
+//	        │                   └── <hex digest>
+//	        │                       └── link
+//	        └── _uploads
+//	            └── <id>
+//	                ├── data
+//	                ├── hashstates
+//	                │   └── <algorithm>
+//	                │       └── <offset>
+//	                └── startedat
```
Comment (deleteriousEffect) [inline registry/storage/paths.go:52]: Was this the tool? That's seriously impressive!

**Review by deleteriousEffect — CHANGES_REQUESTED** [review]
I saw one little thing I'd like to keep like the original, but overall these are reasonable.

File: registry/storage/driver/s3-aws/s3.go (hunk 1 above)
Comment (thaJeztah) [reply registry/storage/driver/s3-aws/s3.go:1178]: This was a tricky one; `go fmt` doesn't accept the indentation on the second line, and formats it as a code block;

```go
// Eg 4 directoryDiff("/path/to/folder/folder1/file", "/path/to/folder/folder2/folder1/file")
//
//        => [ "/path/to/folder/folder2", "/path/to/folder/folder2/folder1" ]
//
```

That felt worse than the original, so I tried to find a somewhat cleaner approach.

Actually, perhaps changing the examples as a whole to a code block would work. Like this;

```go
// directoryDiff finds all directories that are not in common between
// the previous and current paths in sorted order.
//
// # Examples
//
//	directoryDiff("/path/to/folder", "/path/to/folder/folder/file")
//	// => [ "/path/to/folder/folder" ]
//
//	directoryDiff("/path/to/folder/folder1", "/path/to/folder/folder2/file")
//	// => [ "/path/to/folder/folder2" ]
//
//	directoryDiff("/path/to/folder/folder1/file", "/path/to/folder/folder2/file")
//	// => [ "/path/to/folder/folder2" ]
//
//	directoryDiff("/path/to/folder/folder1/file", "/path/to/folder/folder2/folder1/file")
//	// => [ "/path/to/folder/folder2", "/path/to/folder/folder2/folder1" ]
//
//	directoryDiff("/", "/path/to/folder/folder/file")
//	// => [ "/path", "/path/to", "/path/to/folder", "/path/to/folder/folder" ]
```


The function is not exported, but exporting it and rendering the docs then also looks nice;

Screenshot 2022-11-03 at 22 47 22

File: registry/storage/paths.go (hunk 2 above)
Comment (thaJeztah) [reply registry/storage/paths.go:52]: LOL, no this one was some manual elbow-grease 😂 🤷‍♂️ . It did an "ok" job on formatting it (using tabs for indentation), but I still found it hard to read. Given that it was describing a "tree(ish)" output, I thought I'd just use `tree` to do just that.

```bash
mkdir -p "<root>/v2/repositories/<name>/_manifests/revisions/<manifest digest path>"
touch "<root>/v2/repositories/<name>/_manifests/revisions/<manifest digest path>/link"
mkdir -p "<root>/v2/repositories/<name>/_manifests/tags/<tag>/current"
touch "<root>/v2/repositories/<name>/_manifests/tags/<tag>/current/link"
mkdir -p "<root>/v2/repositories/<name>/_manifests/tags/<tag>/index/<algorithm>/<hex digest>"
touch "<root>/v2/repositories/<name>/_manifests/tags/<tag>/index/<algorithm>/<hex digest>/link"
mkdir -p "<root>/v2/repositories/<name>/_layers/<layer links to blob store>"
mkdir -p "<root>/v2/repositories/<name>/_uploads/<id>/hashstates/<algorithm>"
touch "<root>/v2/repositories/<name>/_uploads/<id>/hashstates/<algorithm>/<offset>"
touch "<root>/v2/repositories/<name>/_uploads/<id>/data"
touch "<root>/v2/repositories/<name>/_uploads/<id>/startedat"

tree
```

File: registry/storage/driver/oss/oss_test.go [hunk 3]
```diff
@@ -24,15 +24,18 @@ var ossDriverConstructor func(rootDirectory string) (*Driver, error)
...
-	bucket := os.Getenv("OSS_BUCKET")
-	region := os.Getenv("OSS_REGION")
-	internal := os.Getenv("OSS_INTERNAL")
-	encrypt := os.Getenv("OSS_ENCRYPT")
-	secure := os.Getenv("OSS_SECURE")
-	endpoint := os.Getenv("OSS_ENDPOINT")
-	encryptionKeyID := os.Getenv("OSS_ENCRYPTIONKEYID")
+	var (
+		accessKey       = os.Getenv("ALIYUN_ACCESS_KEY_ID")
+		secretKey       = os.Getenv("ALIYUN_ACCESS_KEY_SECRET")
```
Comment (thaJeztah) [inline registry/storage/driver/oss/oss_test.go:29]: There ones it did automatically

File: registry/storage/driver/swift/swift_test.go [hunk 4]
```diff
@@ -24,48 +24,29 @@ var swiftDriverConstructor func(prefix string) (*Driver, error)
...
+		containerKey, _       = strconv.ParseBool(os.Getenv("SWIFT_TEMPURL_CONTAINERKEY"))
+		tempURLMethods        = strings.Split(os.Getenv("SWIFT_TEMPURL_METHODS"), ",")
 
 		swiftServer *swifttest.SwiftServer
 		err         error
 	)
-	username = os.Getenv("SWIFT_USERNAME")
-	password = os.Getenv("SWIFT_PASSWORD")
-	authURL = os.Getenv("SWIFT_AUTH_URL")
-	tenant = os.Getenv("SWIFT_TENANT_NAME")
```
Comment (thaJeztah) [inline registry/storage/driver/swift/swift_test.go:53]: This one I did manually to match the other drivers; the separate `var` declaration, followed by assigning each variable didn't make a lot of sense, so I combined them

**Comment by thaJeztah** [comment]
@deleteriousEffect updated; PTAL

File: registry/storage/driver/s3-aws/s3.go (hunk 1 above)
Comment (deleteriousEffect) [reply registry/storage/driver/s3-aws/s3.go:1178]: Perfect!

**Review by deleteriousEffect — APPROVED** [review]

**Review by null7073 — COMMENTED** [review]
Of work bot

**Pull request merged**